src/main/java/org/codeforamerica/shiba/pages/PageController.java

```diff
@@ -575,6 +544,26 @@ ModelAndView postFormPage(
     }
   }
 
+  private void updateApplicationStatuses() {
+    if (applicationData.isCAFApplication()) {
+      applicationRepository.updateStatus(applicationData.getId(), CAF, IN_PROGRESS);
+    } else {
+      applicationRepository.updateStatusToNull(CAF, applicationData.getId());
+    }
+    if (applicationData.isCCAPApplication()) {
+      applicationRepository.updateStatus(applicationData.getId(), CCAP, IN_PROGRESS);
+    } else {
+      applicationRepository.updateStatusToNull(CCAP, applicationData.getId());
+    }
+    // TODO - flag?
+    List<String> programs = getValues(applicationData.getPagesData(), APPLICANT_PROGRAMS);
+    if (programs.contains(Program.CERTAIN_POPS)) {
+      applicationRepository.updateStatus(applicationData.getId(), CERTAIN_POPS, IN_PROGRESS);
+    } else {
+      applicationRepository.updateStatusToNull(CERTAIN_POPS, applicationData.getId());
+    }
```

Comment:
Just a general observation that we're making 3 separate db calls here.. probably not a huge issue but it could all be done in one go.

Reply:
Hmm that's a good callout